src/thor/isochrone.cc

```diff
@@ -88,6 +76,28 @@ void Isochrone::ConstructIsoTile(
     }
   }
 
+  max_seconds_ = max_minutes * 60.0f;
+  float max_distance = max_km * 1000.0f;
+  float max_time_est, max_distance_est;
+
+  if (multimodal) {
+    max_time_est = max_distance / (70.0f * kMPHtoMetersPerSec);
+    max_distance_est = max_seconds_ * 70.0f * kMPHtoMetersPerSec;
+  } else if (mode == TravelMode::kPedestrian) {
+    max_time_est = max_distance / (5.0f * kMPHtoMetersPerSec);
+    max_distance_est = max_seconds_ * 5.0f * kMPHtoMetersPerSec;
+  } else if (mode == TravelMode::kBicycle) {
+    max_time_est = max_distance / (5.0f * kMPHtoMetersPerSec);
+    max_distance_est = max_seconds_ * 5.0f * kMPHtoMetersPerSec;
+  } else {
+    // A driving mode
+    max_time_est = max_distance / (5.0f * kMPHtoMetersPerSec);
+    max_distance_est = max_seconds_ * 70.0f * kMPHtoMetersPerSec;
+  }
+
+  max_distance = std::max(max_distance_est, max_distance);
+  max_seconds_ = std::max(max_seconds_, max_time_est);
+
```

Comment:
here I also estimate `max_seconds` similar to how it's been done before for `maxdistance`. I know this is sorta stupid, it's rather a side-effect of what's wrong with how we're currently doing it. `max_seconds_` is the criteria for the dijkstra to stop expanding. So it can't be 0 (as would be the case if there was no `time` contour member at all).